Make WakeLockRequestOptions' signal non-nullable, do not set a default

rakuco opened the pull request:
In Web IDL, the following `WakeLockRequestOptions` instances in JavaScript
should have the same effect (a dictionary that does _not_ have the `signal`
member present):

    {}
    {signal: undefined}

which are different from

    {signal: null}

which means a `WakeLockRequestOptions` dictionary with one member, `signal`,
whose value is `null`.

Setting a default value to the nullable member `signal` means we'd always
have a `signal` member in the dictionary even when we do not set it at
all (first 2 cases), which is not needed and does not make sense.

Additionally, `signal` does not need to be nullable to begin with: by
default, dictionary members do not need to be present, so just make it a
regular, non-nullable member that can either not be present or be set to an
AbortSignal, but not null.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/rakuco/wake-lock/pull/204.html" title="Last updated on May 23, 2019, 11:30 AM UTC (fc328c3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wake-lock/204/ad4afbd...rakuco:fc328c3.html" title="Last updated on May 23, 2019, 11:30 AM UTC (fc328c3)">Diff</a>

rakuco commented:
cc @marcoscaceres

kenchris reviewed (commented):
Better fix this when my other patch lands, or this will just force me to rebases and replace these changes :-( and I will be travelling the next weeks

marcoscaceres commented on index.html:
```diff
@@ -586,11 +586,12 @@ <h3>
-          reject |promise| with an "{{AbortError}}" {{DOMException}},
-          and run <a>abort the request steps</a>.
+          <li>If |options|' |signal:AbortSignal| member is <a>present</a> and its
+          <a>aborted flag</a> is set, then reject |promise| with an "{{AbortError}}"
```
Just a nit, split this into two sub-steps: 
1. reject. 
2. run abort.

rakuco replied:
Done.

marcoscaceres reviewed (approved):
Approved % nit.

rakuco commented:
> Better fix this when my other patch lands, or this will just force me to rebases and replace these changes :-( and I will be travelling the next weeks

Either that or you can grant me write access to your fork so I can rebase #201 on top of this.

Commit:
Make WakeLockRequestOptions' signal non-nullable, do not set a default

In Web IDL, the following `WakeLockRequestOptions` instances in JavaScript
should have the same effect (a dictionary that does _not_ have the `signal`
member present):

    {}
    {signal: undefined}

which are different from

    {signal: null}

which means a `WakeLockRequestOptions` dictionary with one member, `signal`,
whose value is `null`.

Setting a default value to the nullable member `signal` means we'd always
have a `signal` member in the dictionary even when we do not set it at
all (first 2 cases), which is not needed and does not make sense.

Additionally, `signal` does not need to be nullable to begin with: by
default, dictionary members do not need to be present, so just make it a
regular, non-nullable member that can either not be present or be set to an
AbortSignal, but not null.

rakuco commented:
Patch rebased. Please take another look.
